getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,259 @@ Freeing resources
+    but only Resources are reference-counted. This means that if a Dictionary is passed
+    between objects, and the first object is deleted, it will invalidate the second
+    object's variable. This is not so for Resources which won't free their memory until
+    *all* instances are gone.
+
+    .. tabs::
+      .. code-tab:: gdscript GDScript
+
+        extends Node
+
+        class MyObject:
+            extends Object
+            var dict = {}
+
+        func _ready():
+            var obj1 = MyObject.new()
+            var obj2 = MyObject.new()
+            obj1.dict.greeting = "hello"
+            obj2.dict = obj1.dict             # obj2.dict now references obj1's Dictionary
+            obj1.free()                       # obj1 is freed and the Dictionary too!
+            print(obj2.dict.greeting)         # Error! 'greeting' index accessed on null instance!
+
+            # to avoid this, we must manually duplicate the Dictionary
+            obj1 = MyObject.new()
+            obj1.dict.greeting = "hello"
+            obj2.dict = obj1.dict.duplicate() # now we are passing a copy, not a reference
+            obj1.free()                       # obj2's Dictionary still exists
+            print(obj2.dict.greeting)         # prints 'hello'
+
+Godot makes it easy to create custom Resources in the Inspector.
+
+1. Create a plain Resource object in the Inspector. This can even be a type that derives Resource, so long as your script is extending that type.
+2. Set the ``script`` property in the Inspector to be your script.
+
+The Inspector will now display your Resource script's custom properties. If one edits
+those values and saves the resource, the Inspector serializes the custom properties
+too! To save a resource from the Inspector, click the Inspector's tools menu (top right),
+and select "Save" or "Save As...".
+
+If the script's language supports `script classes <https://godot.readthedocs.io/en/latest/getting_started/step_by_step/scripting_continued.html#register-scripts-as-classes>`__, 
+then it streamlines the process. Defining a name for your script alone will add it to
+the Inspector's creation dialog. This will auto-add your script to the Resource
+object you create.
+
+Let's see some examples.
+
+.. tabs::
+  .. code-tab:: gdscript GDScript
+
+    # bot_stats.gd
+    extends Resource
+    export(int) var health
+    export(Resource) var sub_resource
+    export(Array, String) var str_array
+
+    func _init(p_health = 0, p_sub_resource = null, p_str_array = []):
+        health = p_health
+        sub_resource = p_sub_resource
+        str_array = p_str_array
+
+    # bot.gd
+    extends KinematicBody
+    export(Resource) var stats
+
+    const BotStats = preload("res://bot_stats.gd")
+
+    func _init():
+        if not stats:
+            stats = BotStats.new(10)
+    
+    func _ready():
+        print(stats.health) # prints '10'
+  .. code-tab:: csharp
```

Comment:
Is this code-tab's code indented more than the others?

Reply:
Should be fixed now.